core-kotlin-modules/core-kotlin-date-time/src/test/kotlin/com/baeldung/timeconversion/TimeConversionUnitTest.kt

```diff
@@ -0,0 +1,62 @@
+package com.baeldung.timeconversion
+
+import org.junit.jupiter.api.Test
+import kotlin.test.assertNotNull
+import kotlin.test.assertEquals
+
+import kotlinx.datetime.Instant
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.toLocalDateTime
+import java.time.Duration
+import java.util.concurrent.TimeUnit
+
+
+class TimeConversionUnitTest {
+
+
+    @Test
+    fun `given milliseconds, when convert to minutes and seconds using arithmetic operations, should get correct minutes and seconds`() {
+        val milliseconds: Long = 123456L
+        val minutes = milliseconds / 60000
+        val seconds = (milliseconds % 60000) / 1000
+        assertNotNull(minutes)
+        assertNotNull(seconds)
+        assertEquals(2, minutes)
+        assertEquals(3, seconds)
+    }
+
+    @Test
+    fun `given epoch milliseconds, when convert to minutes and seconds using kotlinx datetime, should get correct minutes and seconds`() {
+        val epochMilliseconds: Long = 1699786249000L
+
+        val instant = Instant.fromEpochMilliseconds(epochMilliseconds)
+        val localDateTime = instant.toLocalDateTime(TimeZone.UTC)
+
+        val minutes = localDateTime.hour * 60 + localDateTime.minute.toLong()
+        val seconds = localDateTime.second.toLong()
+
+        assertNotNull(minutes)
+        assertNotNull(seconds)
+        assertEquals(650, minutes)
+        assertEquals(49, seconds)
+    }
+
+    @Test
+    fun `given milliseconds, when convert to minutes and seconds using java TimeUnit library, should get correct minutes and seconds`() {
+        val milliseconds = 123456L
+        val minutes = TimeUnit.MILLISECONDS.toMinutes(milliseconds)
+        val seconds = TimeUnit.MILLISECONDS.toSeconds(milliseconds) - TimeUnit.MINUTES.toSeconds(minutes)
+        assertEquals(2, minutes)
+        assertEquals(3, seconds)
+    }
+
+    @Test
+    fun `given milliseconds, when convert to minutes and seconds using java Duration, should get correct minutes and seconds`() {
+        val milliseconds = 123456L
+        val duration = Duration.ofMillis(milliseconds)
+        val minutes = duration.toMinutes()
+        val seconds = duration.minusMinutes(minutes).seconds
+        assertEquals(2, minutes)
+        assertEquals(3, seconds)
+    }
```

Comment:
SInce this is a Kotlin article, perhaps we should use:
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.time/-duration/

Instead of the Java one?

Reply:
I've revised the PR and draft. We're now using Kotlin-specific classes only